src/services/logging/Logger.ts

```diff
@@ -1,40 +1,152 @@
-import { HostProvider } from "@/hosts/host-provider"
+import * as fs from "fs"
+import * as path from "path"
 import { ErrorService } from "../error"
+import { getVSCodeLogsDir } from "./constants"
+import { formatLogFilenameTimestamp, formatLogMessageTimestamp } from "./timestamp"
 
 /**
  * Simple logging utility for the extension's backend code.
+ * In VS Code mode, logs to console and a file.
+ * In standalone mode, logs to console (which is redirected to file by the parent process).
  */
 export class Logger {
-	public readonly channelName = "Cline Dev Logger"
+	private static fileStream?: fs.WriteStream
+	private static logFilePath?: string
+
+	// TODO (celestial-vault): This does not handle the case where log files are deleted while the logger
+	// is active (either manually by the user or via clearLogs()). The WriteStream will
+	// continue writing to the unlinked file descriptor, which means logs appear to work
+	// but are not actually being persisted to disk. A proper solution would need to handle
+	// effectively checking for the existence of the log file before writing to it.
+	/**
+	 * Ensures log file is ready for writing. Creates a new log file if needed.
+	 * This method is called automatically on first use.
+	 */
+	private static ensureLogFileReady(): void {
+		// Skip file logging in standalone mode - console is redirected by parent process
+		if (process.env.IS_STANDALONE === "true") {
+			return
+		}
+
+		if (Logger.fileStream) {
+			return
+		}
+
+		try {
+			const logsDir = getVSCodeLogsDir()
+			if (!fs.existsSync(logsDir)) {
+				fs.mkdirSync(logsDir, { recursive: true })
+			}
+
+			const timestamp = formatLogFilenameTimestamp()
+			const pid = process.pid
+			const logFileName = `cline-vscode-${timestamp}-${pid}.log`
+
+			Logger.logFilePath = path.join(logsDir, logFileName)
+			Logger.fileStream = fs.createWriteStream(Logger.logFilePath, { flags: "a" })
+
+			console.log(`Logger initialized - logs will be written to: ${Logger.logFilePath}`)
+		} catch (error) {
+			console.error("Failed to initialize file logging:", error)
+		}
+	}
+
+	/**
+	 * Clean up file logging resources.
+	 * Should be called in extension deactivate().
+	 */
+	static cleanup(): void {
```

Comment:
Consider using the dispose pattern and marrying this to VSCode-style disposers in the extension registration. Probably clearer to the casual reader what dispose does versus something called cleanup.

If we hate dispose, that's fine, I'm a tourist.